ush/gfs/task/analysis.py

```diff
@@ -0,0 +1,409 @@
+
+    def finalize(self):
+        super().finalize()
+
+    def get_obs_dict(self):
+        """
+        Get a dictionary of observation files to copy/use
+        based on the specified configuration
+        """
+        import ufsda # temporary until this is in workflow
+        import yaml
+        with open(self.obs_list_yaml, 'r') as yamlopen:
+            obs_list_dict = yaml.safe_load(yamlopen)
+        # need to replace vars
+        obs_list_dict['OBS_YAML_DIR'] = self.obs_yaml_dir
+        obs_list_dict = ufsda.yamltools.replace_vars(obs_list_dict)
+        del obs_list_dict['OBS_YAML_DIR']
+        # need a few extra variables defined
+        obs_list_dict['OBS_DATE'] = self.cdate
+        obs_list_dict['OBS_DIR'] = os.path.join(self.datadir, 'obs')
+        obs_list_dict['OBS_PREFIX'] = os.environ['OPREFIX']
+        obs_list_dict = ufsda.yamltools.parse_config(obs_list_dict)
+        # get observers from master dictionary
+        observers = obs_list_dict['observers']
+        obs_dict = {}
+        for ob in observers:
+            obfile = ob['obs space']['obsdatain']['engine']['obsfile']
+            basename = os.path.basename(obfile)
+            obs_dict[os.path.join(os.environ['COMIN_OBS'], basename)] = obfile
+        return obs_dict
+
+    def get_staticb_dict(self):
+        """
+        get dictionary of staticb files to copy/use
+        based on the specified configuration.
+        """
+        import ufsda # temporary until this is in workflow
+        import yaml
+        with open(self.berror_yaml, 'r') as yamlopen:
+            berror_yaml_dict = yaml.safe_load(yamlopen)
+
+    def stage_obs(self, filedict):
+        logging.info('Staging observations')
+        self.stage(filedict, skip_missing=True)
+        logging.info('Finished staging observations')
+
+    def stage_bkg(self, filedict):
+        logging.info('Staging model backgrounds')
+        self.stage(filedict)
+        logging.info('Finished staging backgrounds')
+
+    def stage_fix(self, filedict):
+        logging.info('Staging fix files')
+        self.stage(filedict)
+        logging.info('Finished staging fix files')
+
+    def stage_berror(self, filedict):
+        logging.info('Staging staticb files')
+        self.stage(filedict)
+        logging.info('Finished staging staticb files')
+
+    def stage(self, filedict, skip_missing=False):
+        for src, dest in filedict.items():
+            destdir = os.path.dirname(dest)
+            if not os.path.exists(destdir):
+                logging.info(f'{destdir} does not exist, creating directory.')
+                os.makedirs(destdir)
+            if os.path.exists(dest):
+                os.remove(dest)
+            if skip_missing:
+                if not os.path.exists(src):
+                    logging.warning(f'{src} does not exist. Will not copy.')
+                    continue
+            logging.info(f'Copying {src} to {dest}')
+            shutil.copyfile(src, dest)
```

Comment:
I think this answers my question above.   `stage` copies files.

Reply:
But this is a good point, we could easily add something that is like so:
`export STAGE_LN="YES"` to a config. file and then have the stage method take that as an optional argument. I propose we explore this in a future PR after discussion with @aerorahul

Reply:
Sounds good to me.